src/client/datascience/kernel-launcher/localKnownPathKernelSpecFinder.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { CancellationToken, Uri } from 'vscode';
+import { IWorkspaceService } from '../../common/application/types';
+import { IFileSystem } from '../../common/platform/types';
+import { getKernelId } from '../jupyter/kernels/helpers';
+import { KernelSpecConnectionMetadata, PythonKernelConnectionMetadata } from '../jupyter/kernels/types';
+import { IJupyterKernelSpec } from '../types';
+import { LocalKernelSpecFinderBase } from './localKernelSpecFinderBase';
+import { JupyterPaths } from './jupyterPaths';
+import { PYTHON_LANGUAGE } from '../../common/constants';
+
+/**
+ * This class searches for kernels on the file system in well known paths documented by Jupyter.
+ * This will return Python, Julia, R etc kernels.
+ * Returns all kernels regardless of whether Python extension is installed or not.
+ */
+@injectable()
+export class LocalKnownPathKernelSpecFinder extends LocalKernelSpecFinderBase {
+    constructor(
+        @inject(IFileSystem) fs: IFileSystem,
+        @inject(IWorkspaceService) workspaceService: IWorkspaceService,
+        @inject(JupyterPaths) private readonly jupyterPaths: JupyterPaths
+    ) {
+        super(fs, workspaceService);
+    }
+    /**
+     * @param {boolean} includePythonKernels Include/exclude Python kernels in the result.
+     */
+    public async listKernelSpecs(
+        includePythonKernels: boolean,
+        cancelToken?: CancellationToken
+    ): Promise<(KernelSpecConnectionMetadata | PythonKernelConnectionMetadata)[]> {
+        return this.listKernelsWithCache(includePythonKernels ? 'IncludePython' : 'ExcludePython', async () => {
```

Comment:
So the cache key can just be a constant here since it's the global search? That make sense.